### PR TITLE
Spi slave mode

### PR DESCRIPTION
**Pull request by apeng2012**

### CODE REVIEW

**Review by burrbull — APPROVED** [review]

**Comment by burrbull** [comment]
Nice!

**Comment by apeng2012** [comment]
When the length of the transferred data is variable, DMA cannot be used, but an interrupt method is used.

File: src/spi.rs [hunk 1]
```diff
@@ -135,6 +164,27 @@ remap!(Spi3NoRemap, pac::SPI3, false, PB3, PB4, PB5);
...
+}
+
+remap_slave!(Spi1NoRemapSlave, pac::SPI1, false, PA5, PA6, PA7);
+remap_slave!(Spi1RemapSlave, pac::SPI1, true, PB3, PB4, PB5);
+remap_slave!(Spi2NoRemapSlave, pac::SPI2, false, PB13, PB14, PB15);
+#[cfg(any(feature = "high", feature = "connectivity"))]
+remap_slave!(Spi3NoRemapSlave, pac::SPI3, false, PB3, PB4, PB5);
+#[cfg(feature = "connectivity")]
+remap_slave!(Spi3RemapSlave, pac::SPI3, true, PC10, PC11, PC12);
+
```
Comment (burrbull) [inline src/spi.rs:187]: What sense of duplicating?

Comment (burrbull) [reply src/spi.rs:187]: https://github.com/stm32-rs/stm32f1xx-hal/commit/bb248dff6ce4271feb7248f006a341b3490e0c8a

Comment (burrbull) [reply src/spi.rs:187]: Don't see any changes in PR

Comment (apeng2012) [reply src/spi.rs:187]: Oh， Still need PR

Comment (burrbull) [reply src/spi.rs:187]: just merge or cherry-pick commit in this branch if all works as expected

**Comment by apeng2012** [comment]
Overrun error are about receivy, not send

**Comment by burrbull** [comment]
merging. Please open issue and describe errors if present.
bors r+

**Pull request merged**